pyrometheus/codegen/python.py

```diff
@@ -286,6 +293,69 @@ def get_mixture_internal_energy_mass(self, temperature, mass_fractions):
         emix = self.get_mass_average_property(mass_fractions, e0_rt)
         return self.gas_constant * temperature * emix
 
+    def get_species_viscosities(self, temperature):
+        return self.usr_np.sqrt(temperature)*self._pyro_make_array([
+                % for sp in range(sol.n_species):
+                ${cgm(ce.transport_polynomial_expr(sol.get_viscosity_polynomial(sp),
+                                                    2, Variable("temperature")))},
+                % endfor
+                ])
+
+    def get_mixture_viscosity_mixavg(self, temperature, mass_fractions):
+        mole_fractions = self.iwts * mass_fractions\
+ * self.get_mix_molecular_weight(mass_fractions)
+        viscosities = self.get_species_viscosities(temperature)
+        mix_rule_f = self._pyro_make_array([
+            %for sp in range(sol.n_species):
+            ${cgm(ce.viscosity_mixture_rule_wilke_expr(sol, sp,
+                 Variable("mole_fractions"), Variable("viscosities")))},
+            %endfor
+            ])
+        return sum(mole_fractions*viscosities/mix_rule_f)
+
+    def get_species_thermal_conductivities(self, temperature):
+        return self.usr_np.sqrt(temperature)*self._pyro_make_array([
+                % for sp in range(sol.n_species):
+                ${cgm(ce.transport_polynomial_expr(
+                    sol.get_thermal_conductivity_polynomial(sp), 1,
+                    Variable("temperature")))},
+                % endfor
+                ])
+
+    def get_mixture_thermal_conductivity_mixavg(self, temperature, mass_fractions):
+        mole_fractions = self.iwts * mass_fractions\
+ * self.get_mix_molecular_weight(mass_fractions)
+        conductivities = self.get_species_thermal_conductivities(temperature)
+        return 0.5*(sum(mole_fractions*conductivities)
+            + 1/sum(mole_fractions/conductivities))
+
+    def get_species_binary_mass_diffusivities(self, temperature):
+        return self._pyro_make_array([
+                % for i, j, in product(range(sol.n_species), range(sol.n_species)):
+                ${cgm(ce.transport_polynomial_expr(
+                      sol.get_binary_diff_coeffs_polynomial(i, j), 1,
+                      Variable("temperature")))},
+                % endfor
+                ]).reshape((self.num_species, self.num_species))
```

Comment:
Will this shape work in both the array-of-values-on-grid (via object arrays) and the 0D reactor (all species in one numpy array) case?

Reply:
@ecisneros8 , based on your experience with grids, what do you think?

Reply:
The 0D reactor is being exercised with the new test functions. Now, if you are referring to grids as the Mirge-COM case, for instance, it is working with the current implementation. A PR is pending (https://github.com/illinois-ceesd/mirgecom/pull/715) where it exercises this capability (plus, I've been using this code in different cases over the last months).